staging/src/k8s.io/client-go/tools/events/helpers.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/api/events/v1beta1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GetFieldSelector returns the appropriate field selector based on the API version being used to communicate with the server.
+// The returned field selector can be used with List and Watch to filter desired events.
+func GetFieldSelector(groupVersion schema.GroupVersion, regardingName, regardingNamespace, regardingKind, regardingUID string) (fields.Selector, error) {
+	field := fields.Set{}
+	var err error = nil
+
+	switch groupVersion {
+	case v1beta1.SchemeGroupVersion:
```

Comment:
The structure of this method looks reasonable, but does the v1beta1 field selector actually work? I'm not seeing where the `regarding...` field selectors are defined for v1beta1 events.

The internal map of fields the selector can use is built here:

https://github.com/kubernetes/kubernetes/blob/81a8b9804a3bf310cb74446bd8a8e08e0317de22/pkg/registry/core/event/strategy.go#L99-L115

And the transformer from v1 field selector keys to internal keys used for mapping is here:

https://github.com/kubernetes/kubernetes/blob/81a8b9804a3bf310cb74446bd8a8e08e0317de22/pkg/apis/core/v1/conversion.go#L448-L469

But I'm not seeing a similar transformer for v1beta1 events (which would need to map the v1beta1 fieldSelector key "regarding.name" to "involvedObject.name" internally for matching, etc)

Reply:
IC, I will supplement this part, thanks.

Reply:
@liggitt I added the conversion function.